### PR TITLE
Fixes improvements for docker compose and terraform quickstarts

### PR DESCRIPTION
**Pull request by douglaswainer**
## Description

The primary purpose of this PR is to make the docker-compose quickstart easier to get working out of the box for people trying to run it on bare metal machines with NVME drives. It also addresses the some issues I encountered when testing these changes with the other docker-compose based quickstart guides.
 - Docker-compose (on bare metal)
 - Terraform Equinix metal
 - Vagrant (with USE_HELM=false

Plus some cleanup to remove some code/files that I believe are legacy and no longer used. 

## Why is this needed

This is my attempt to address [172](https://github.com/tinkerbell/sandbox/issues/172) and [173](https://github.com/tinkerbell/sandbox/issues/173). 

## How Has This Been Tested?

I've tested by running through the quickstart documentation for all compose based quickstarts:
 - Docker-compose (On bare metal Lenovo Thinkcenter Tiny M900 machines, with NVME drives)
 - Terraform
 - Vagrant/Libvirt with USE_HELM=true
 - Vagrant/Libvirt with USE_HELM=false
 
## How are existing users impacted? What migration steps/scripts do we need?

All statements like `{{ index .Hardware.Disks 0 }}` in the deploy/stack/compose/manifests/template.yaml have been replaced with environment variables (set in the .env file) that get substituted by the [ `manifest-update`](https://github.com/tinkerbell/sandbox/blob/47cfd6d0a0b659f1e364a78a4e63e08cdf168ca8/deploy/stack/compose/docker-compose.yml#LL193C3-L193C18) container task. This should make it more likely to work for new users, but might confuse users that add their own manifest files. 

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] added unit or e2e tests
- [x] provided instructions on how to upgrade

### CODE REVIEW

File: deploy/infrastructure/terraform/.env [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Can be set to your own hook builds
+vOSIE=v0.7.0
```
Comment (jacobweinstock) [inline deploy/infrastructure/terraform/.env:2]: mind using the v0.8.0 version here?

File: deploy/stack/compose/manifests/template.yaml [hunk 2]
```diff
@@ -19,23 +19,23 @@ spec:
             image: quay.io/tinkerbell-actions/image2disk:v1.0.0
             timeout: 600
             environment:
-              DEST_DISK: {{ index .Hardware.Disks 0 }}
+              DEST_DISK: $DISK_DEVICE
```
Comment (jacobweinstock) [inline deploy/stack/compose/manifests/template.yaml:22]: im curious, why the move to an env var over using the value in the hardware manifest?

File: deploy/stack/compose/setup.sh [hunk 3]
```diff
@@ -2,10 +2,11 @@
 
 install_docker() {
 	curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
-	add-apt-repository "deb https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+	add-apt-repository --yes "deb https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
 	update_apt
 	apt-get install --no-install-recommends containerd.io docker-ce docker-ce-cli docker-compose-plugin
 	gpasswd -a vagrant docker
+	apt-get install docker-compose
```
Comment (jacobweinstock) [inline deploy/stack/compose/setup.sh:9]: i believe `docker-ce`/`docker-ce-cli` come with the `docker compose` command. is the older `docker-compose` needed?

File: deploy/stack/compose/setup.sh [hunk 4]
```diff
@@ -87,19 +57,12 @@ tweak_bash_interactive_settings() {
...
 	update_apt
 	install_docker
 	install_kubectl
 
-	# setup_layer2_network "$host_ip"
-
-	setup_compose_env_overrides "$host_ip" "$worker_ip" "$worker_mac" "$compose_dir"
-	docker compose --env-file "$compose_dir"/.env -f "$compose_dir"/docker-compose.yml up -d
+	local compose_dir="/sandbox/stack/compose"
+	docker-compose -f "$compose_dir"/docker-compose.yml up -d
```
Comment (jacobweinstock) [inline deploy/stack/compose/setup.sh:65]: im curious why the move to the older `docker-compose` command over the newer `docker compose` command?

**Review by jacobweinstock — COMMENTED** [review]
Tremendous work, @douglaswainer, thank you!

**Comment by jacobweinstock** [comment]
> I've been looking into the [hardware types](https://github.com/tinkerbell/tink/blob/main/api/v1alpha1/hardware_types.go) and I can see how I could have used the `MetadataInstanceStorageMount`, which I believe could be substituted more naturally (or more inline with Tinkerbell templates) with:
> 
> ```
> {{ (index .Hardware.Metadata.Instance.Storage.Filesystems 0).Mount.Device }}
> ```
> 
> Provided the `hardware.yaml` provides this metadata. I might be wrong, I'm not very familiar with Golang templates so I'll need to test.
> 
> I'm happy to close this and go and rework/retest my changes and then reopen again at a later stage, if that is preferable?

Hey @douglaswainer, at the moment only the disks (`hardware.Spec.Disks`) are available for use like this. This was deliberate on our part so that we could evolve this functionality with purpose as use-cases arise. So no need to rework this. Thanks though!

File: deploy/infrastructure/terraform/.env (hunk 1 above)
Comment (douglaswainer) [reply deploy/infrastructure/terraform/.env:2]: Absolutely! I'll upgrade and run through my tests again :+1:

File: deploy/stack/compose/setup.sh (hunk 3 above)
Comment (douglaswainer) [reply deploy/stack/compose/setup.sh:9]: This is just an old, bad habit of mine. You are quite right, I'll fix this!

File: deploy/stack/compose/setup.sh (hunk 4 above)
Comment (douglaswainer) [reply deploy/stack/compose/setup.sh:65]: It's just an old, bad habit of mine. I'll fix it!

**Comment by douglaswainer** [comment]
> Hey @douglaswainer, at the moment only the disks ([hardware.Spec.Disks](https://github.com/tinkerbell/tink/blob/9c098c135ff221757e9baa4116900600dbeefe07/pkg/controllers/workflow/controller.go#L137)) are available for use like this. This was deliberate on our part so that we could evolve this functionality with purpose as use-cases arise. So no need to rework this. Thanks though!

Oh interesting, I assumed the whole hardware data json could be referenced. Thank you for clarifying!

File: deploy/stack/compose/manifests/template.yaml (hunk 2 above)
Comment (jacobweinstock) [reply deploy/stack/compose/manifests/template.yaml:22]: this is great feedback, thank you! im good with this.

File: deploy/infrastructure/terraform/.env [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Can be set to your own hook builds
```
Comment (jacobweinstock) [inline deploy/infrastructure/terraform/.env:1]: im not too excited about duplicate `.env` files to maintain. any chance we could have only one?

Comment (douglaswainer) [reply deploy/infrastructure/terraform/.env:1]: Yeah I'm not happy about it either, I can look into options for merging them into one but I don't know of an easy solution based on .env files. 

There's a few bits of reused variables and bash code in the project, I was thinking of converting the bash provisioning to Ansible and splitting out the more specific provisioning tasks with roles based on whether the machine is bare metal, equinix, vagrant etc.. Though I think that would be a separate issue/PR for later on.

Comment (jacobweinstock) [reply deploy/infrastructure/terraform/.env:1]: thanks for the response and especially for your work and patience on this!

**Review by jacobweinstock — APPROVED** [review]

**Pull request merged**